src/visualizations/config/adapters/dhis_highcharts/index.js

```diff
@@ -170,7 +170,7 @@ export default function ({ store, layout, el, extraConfig, extraOptions }) {
     }
 
     // hide empty categories
-    if (_layout.hideEmptyRowItems !== 'NONE') {
+    if (_layout.hideEmptyRowItems && _layout.hideEmptyRowItems !== 'NONE') {
```

Comment:
Consider exporting the valid `hideEmptyRowItems` options and check if this value is actually one of them?

Reply:
🤔 this would be a breaking change then, as it requires a change in DV as well to make it work.

Reply:
This if-check checks for ALL, BEFORE_FIRST, AFTER_LAST, BEFORE_FIRST_AFTER_LAST? So could we make them consts in this repo, then use them here in this check and also here https://github.com/dhis2/analytics/blob/ef877bec63ea3066dbc8b3538f61af409375b726/src/visualizations/config/adapters/dhis_highcharts/getTrimmedConfig.js#L74-L93 ?

Reply:
`hideEmptyRowItems` is not always passed in `_layout` regardless of default/set values.
Is this what you are suggesting?
```suggestion
    if ([ALL, BEFORE_FIRST, AFTER_LAST, BEFORE_FIRST_AFTER_LAST].includes(_layout.hideEmptyRowItems)) {
```

Reply:
Yes

Reply:
Refactored